src/main/java/com/diplomaProject/StudyBe/Subject/Cotroller/SubjectController.java

```diff
@@ -1,10 +1,22 @@
 package com.diplomaProject.StudyBe.Subject.Cotroller;
 
 
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.diplomaProject.StudyBe.Subject.Subject;
+import com.diplomaProject.StudyBe.Subject.web.dto.SubjectDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import com.diplomaProject.StudyBe.Subject.Service.SubjectService;
 
 @RestController
-@RequestMapping("/role")
+@RequestMapping("/subject")
 public class SubjectController {
+
+    @Autowired
+    private SubjectService service;
+    @PostMapping("/addSubject")
+    public void addSubject(@RequestBody SubjectDto subject){
```

Comment:
Add a linter to your editor and lint it often there is a option to do it on save.

Reply:
Also restrain from making the endpoints  extra text  currently there is /subject/addSubject
make the subject on Post request on only /subject 
In future restrain form adding verbs to your endpoints example [deleting, updaring, changing ] and so on.